src/commands/schema/pull.js

```diff
@@ -0,0 +1,85 @@
+const FaunaCommand = require("../../lib/fauna-command.js");
+const fetch = require("node-fetch");
+const fs = require("fs");
+const path = require("path");
+const { Flags } = require("@oclif/core");
+
+class PullSchemaCommand extends FaunaCommand {
+  async run() {
+    const {
+      connectionOptions: { domain, port, scheme, secret },
+    } = await this.getClient();
+
+    const dir = this.flags.dir;
+
+    try {
+      const filesres = await fetch(
+        `${scheme}://${domain}:${port}/schema/1/files`,
+        {
+          method: "GET",
+          headers: { AUTHORIZATION: `Bearer ${secret}` },
+        }
+      );
+      const filesjson = await filesres.json();
+      if (filesjson.error) {
+        this.error(filesjson.error.message);
+      }
+      // Sort for consistent order. It's nice for tests.
+      const filenames = filesjson.files
+        .map((file) => file.filename)
+        .filter((name) => name.endsWith(".fsl"))
+        .sort();
+
+      // All of the below could be parallelized if necessary.
+      if (this.flags.delete) {
+        // Delete all .fsl files (not directories).
+        for (const deleteme of fs.readdirSync(dir)) {
+          const fp = path.join(dir, deleteme);
+          if (deleteme.endsWith(".fsl") && !fs.statSync(fp).isDirectory()) {
+            fs.unlinkSync(fp);
+          }
+        }
+      }
+
+      for (const filename of filenames) {
+        const fileres = await fetch(
+          `${scheme}://${domain}:${port}/schema/1/files/${filename}`,
+          {
+            method: "GET",
+            headers: { AUTHORIZATION: `Bearer ${secret}` },
+          }
+        );
+        const filejson = await fileres.json();
+        if (filejson.error) {
+          this.error(filejson.error.message);
+        }
+        fs.writeFileSync(path.join(dir, filename), filejson.content);
```

Comment:
I think this needs a `fs.mkdirSync(path.parent(path.join(dir, filename)), { recursive: true })` to make the parent directories for these files

Reply:
Yeah, good point. The dir given might not exist yet.

Reply:
Actually, I take it back. The user should make the directory. They should set the permissions, etc.

Reply:
I could see us taking a dir and making a subdirectory of that directory for all the files, but not making the top-level directory ourselves. Some extension of the tool might do something like that.

Reply:
Note from standup: Neil's comment above was for subdirectories which we do support.

Regarding the top level, we should do what rsync does.